src/infrastructure/fragmenter.rs

```diff
@@ -60,31 +61,20 @@ impl Fragmentation {
     }
 
     /// Split the given payload into fragments and write those fragments to the passed packet data.
-    pub fn spit_into_fragments(
-        payload: &[u8],
-        acked_header: AckedPacketHeader,
-        packet_data: &mut PacketData,
-        config: &Config,
-    ) -> Result<()> {
+    pub fn spit_into_fragments<'a>(payload: &'a [u8], config: &Config) -> Result<Vec<&'a [u8]>> {
+        let mut fragments = Vec::new();
+
         let payload_length = payload.len() as u16;
         let num_fragments =
-            Fragmentation::total_fragments_needed(payload_length, config.fragment_size) as u8; /* safe cast max fragments is u8 */
+            // Safe cast max fragments is u8
```

Comment:
I don't quite understand what this comment is trying to say

Reply:
Not related to this PR, but it means that the maximal fragments are restricted to 256; u8. And thus this cast is safe.